0xProject/0x-monorepo
# Fix/pipeline/ohlcv ratelimit

## Description

Use `Bottleneck` instead of `p-limit` to enforce time-based throttling. We're enforcing a wait of at least 67ms between calls (can be adjusted with env variable) and a hard limit of 1800 calls per minute. 

CC rate limit is 50/s or 2000/min. 

There's a few messy commits here so will be squashing commits before merging. 

## Testing instructions

You can run the script locally at lib/src/scripts/pull_ohlcv_cryptocompare.js.

I haven't been able to run on AWS yet (still need to get an AWS account set up). We should run on airflow to see if it works. If we encounter rate limit errors, try setting `process.env.CRYPTOCOMPARE_MAX_REQS_PER_SECOND = 10`.

## Types of changes



* Bug fix (non-breaking change which fixes an issue)





## Checklist:
